picky-krb dependency build failure

picky-krb causes a build error on a fresh cargo project.

Simple reproducer:

```sh
$ mkdir myproj
$ cd myproj/
$ cargo init
    Creating binary (application) package
note: see more `Cargo.toml` keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
$ cargo add picky-krb
    Updating crates.io index
      Adding picky-krb v0.11.2 to dependencies
    Updating crates.io index
     Locking 53 packages to latest Rust 1.90.0 compatible versions
$ cargo build
  Downloaded const-oid v0.10.1
  Downloaded picky-asn1 v0.10.1
  Downloaded crypto-common v0.2.0-rc.4
  Downloaded serdect v0.4.1
  Downloaded inout v0.2.1
  Downloaded block-buffer v0.11.0-rc.5
  Downloaded picky-asn1-der v0.5.3
  Downloaded byteorder v1.5.0
  Downloaded sha1 v0.11.0-rc.2
  Downloaded cpufeatures v0.2.17
  Downloaded block-padding v0.4.1
  Downloaded digest v0.11.0-rc.3
  Downloaded serde_bytes v0.11.19
  Downloaded pbkdf2 v0.13.0-rc.1
  Downloaded cipher v0.5.0-rc.1
  Downloaded base16ct v0.3.0
  Downloaded oid v0.2.1
  Downloaded hybrid-array v0.4.5
  Downloaded cbc v0.2.0-rc.1
  Downloaded des v0.9.0-rc.1
  Downloaded hmac v0.13.0-rc.2
  Downloaded num-traits v0.2.19
  Downloaded typenum v1.19.0
  Downloaded uuid v1.18.1
  Downloaded picky-krb v0.11.2
  Downloaded picky-asn1-x509 v0.15.0
  Downloaded aes v0.9.0-rc.1
  Downloaded crypto-bigint v0.7.0-rc.8
  Downloaded 28 crates (1.2MiB) in 0.27s
   Compiling typenum v1.19.0
   Compiling proc-macro2 v1.0.101
   Compiling unicode-ident v1.0.19
   Compiling quote v1.0.41
   Compiling serde_core v1.0.228
   Compiling cfg-if v1.0.3
   Compiling serde v1.0.228
   Compiling libc v0.2.176
   Compiling getrandom v0.3.3
   Compiling zerocopy v0.8.27
   Compiling subtle v2.6.1
   Compiling const-oid v0.10.1
   Compiling autocfg v1.5.0
   Compiling cpufeatures v0.2.17
   Compiling thiserror v1.0.69
   Compiling base64 v0.22.1
   Compiling num-traits v0.2.19
   Compiling byteorder v1.5.0
   Compiling syn v2.0.106
   Compiling hybrid-array v0.4.5
   Compiling rand_core v0.9.3
   Compiling crypto-common v0.2.0-rc.4
   Compiling block-padding v0.4.1
   Compiling block-buffer v0.11.0-rc.5
   Compiling inout v0.2.1
   Compiling crypto-bigint v0.7.0-rc.8
   Compiling ppv-lite86 v0.2.21
   Compiling cipher v0.5.0-rc.1
   Compiling digest v0.11.0-rc.3
   Compiling rand_chacha v0.9.0
error[E0432]: unresolved import `inout::block_padding::UnpadError`
  --> /.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/cipher-0.5.0-rc.1/src/block.rs:20:30
   |
20 |     block_padding::{Padding, UnpadError},
   |                              ^^^^^^^^^^ no `UnpadError` in the root

error[E0107]: trait takes 0 generic arguments but 1 generic argument was supplied
  --> /.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/cipher-0.5.0-rc.1/src/block.rs:90:38
   |
90 |     fn encrypt_padded_inout<'out, P: Padding<Self::BlockSize>>(
   |                                      ^^^^^^^----------------- help: remove the unnecessary generics
   |                                      |
   |                                      expected 0 generic arguments
   |
note: trait defined here, with 0 generic parameters
  --> /.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/block-padding-0.4.1/src/lib.rs:16:11
   |
16 | pub trait Padding {
   |           ^^^^^^^

error[E0107]: trait takes 0 generic arguments but 1 generic argument was supplied
   --> /.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/cipher-0.5.0-rc.1/src/block.rs:107:30
    |
107 |     fn encrypt_padded<'a, P: Padding<Self::BlockSize>>(
    |                              ^^^^^^^----------------- help: remove the unnecessary generics
    |                              |
    |                              expected 0 generic arguments
    |
note: trait defined here, with 0 generic parameters
   --> /.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/block-padding-0.4.1/src/lib.rs:16:11
    |
 16 | pub trait Padding {
    |           ^^^^^^^

error[E0107]: trait takes 0 generic arguments but 1 generic argument was supplied
   --> /.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/cipher-0.5.0-rc.1/src/block.rs:121:34
    |
121 |     fn encrypt_padded_b2b<'a, P: Padding<Self::BlockSize>>(
    |                                  ^^^^^^^----------------- help: remove the unnecessary generics
    |                                  |
    |                                  expected 0 generic arguments
    |
note: trait defined here, with 0 generic parameters
   --> /.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/block-padding-0.4.1/src/lib.rs:16:11
    |
 16 | pub trait Padding {
    |           ^^^^^^^

error[E0107]: trait takes 0 generic arguments but 1 generic argument was supplied
   --> /.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/cipher-0.5.0-rc.1/src/block.rs:204:38
    |
204 |     fn decrypt_padded_inout<'out, P: Padding<Self::BlockSize>>(
    |                                      ^^^^^^^----------------- help: remove the unnecessary generics
    |                                      |
    |                                      expected 0 generic arguments
    |
note: trait defined here, with 0 generic parameters
   --> /.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/block-padding-0.4.1/src/lib.rs:16:11
    |
 16 | pub trait Padding {
    |           ^^^^^^^

error[E0107]: trait takes 0 generic arguments but 1 generic argument was supplied
   --> /.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/cipher-0.5.0-rc.1/src/block.rs:222:30
    |
222 |     fn decrypt_padded<'a, P: Padding<Self::BlockSize>>(
    |                              ^^^^^^^----------------- help: remove the unnecessary generics
    |                              |
    |                              expected 0 generic arguments
    |
note: trait defined here, with 0 generic parameters
   --> /.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/block-padding-0.4.1/src/lib.rs:16:11
    |
 16 | pub trait Padding {
    |           ^^^^^^^

error[E0107]: trait takes 0 generic arguments but 1 generic argument was supplied
   --> /.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/cipher-0.5.0-rc.1/src/block.rs:236:34
    |
236 |     fn decrypt_padded_b2b<'a, P: Padding<Self::BlockSize>>(
    |                                  ^^^^^^^----------------- help: remove the unnecessary generics
    |                                  |
    |                                  expected 0 generic arguments
    |
note: trait defined here, with 0 generic parameters
   --> /.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/block-padding-0.4.1/src/lib.rs:16:11
    |
 16 | pub trait Padding {
    |           ^^^^^^^

error[E0107]: trait takes 0 generic arguments but 1 generic argument was supplied
   --> /.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/cipher-0.5.0-rc.1/src/block.rs:341:38
    |
341 |     fn encrypt_padded_inout<'out, P: Padding<Self::BlockSize>>(
    |                                      ^^^^^^^----------------- help: remove the unnecessary generics
    |                                      |
    |                                      expected 0 generic arguments
    |
note: trait defined here, with 0 generic parameters
   --> /.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/block-padding-0.4.1/src/lib.rs:16:11
    |
 16 | pub trait Padding {
    |           ^^^^^^^

error[E0107]: trait takes 0 generic arguments but 1 generic argument was supplied
   --> /.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/cipher-0.5.0-rc.1/src/block.rs:358:26
    |
358 |     fn encrypt_padded<P: Padding<Self::BlockSize>>(
    |                          ^^^^^^^----------------- help: remove the unnecessary generics
    |                          |
    |                          expected 0 generic arguments
    |
note: trait defined here, with 0 generic parameters
   --> /.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/block-padding-0.4.1/src/lib.rs:16:11
    |
 16 | pub trait Padding {
    |           ^^^^^^^

error[E0107]: trait takes 0 generic arguments but 1 generic argument was supplied
   --> /.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/cipher-0.5.0-rc.1/src/block.rs:372:34
    |
372 |     fn encrypt_padded_b2b<'a, P: Padding<Self::BlockSize>>(
    |                                  ^^^^^^^----------------- help: remove the unnecessary generics
    |                                  |
    |                                  expected 0 generic arguments
    |
note: trait defined here, with 0 generic parameters
   --> /.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/block-padding-0.4.1/src/lib.rs:16:11
    |
 16 | pub trait Padding {
    |           ^^^^^^^

error[E0107]: trait takes 0 generic arguments but 1 generic argument was supplied
   --> /.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/cipher-0.5.0-rc.1/src/block.rs:459:38
    |
459 |     fn decrypt_padded_inout<'out, P: Padding<Self::BlockSize>>(
    |                                      ^^^^^^^----------------- help: remove the unnecessary generics
    |                                      |
    |                                      expected 0 generic arguments
    |
note: trait defined here, with 0 generic parameters
   --> /.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/block-padding-0.4.1/src/lib.rs:16:11
    |
 16 | pub trait Padding {
    |           ^^^^^^^

error[E0107]: trait takes 0 generic arguments but 1 generic argument was supplied
   --> /.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/cipher-0.5.0-rc.1/src/block.rs:477:26
    |
477 |     fn decrypt_padded<P: Padding<Self::BlockSize>>(
    |                          ^^^^^^^----------------- help: remove the unnecessary generics
    |                          |
    |                          expected 0 generic arguments
    |
note: trait defined here, with 0 generic parameters
   --> /.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/block-padding-0.4.1/src/lib.rs:16:11
    |
 16 | pub trait Padding {
    |           ^^^^^^^

error[E0107]: trait takes 0 generic arguments but 1 generic argument was supplied
   --> /.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/cipher-0.5.0-rc.1/src/block.rs:491:34
    |
491 |     fn decrypt_padded_b2b<'a, P: Padding<Self::BlockSize>>(
    |                                  ^^^^^^^----------------- help: remove the unnecessary generics
    |                                  |
    |                                  expected 0 generic arguments
    |
note: trait defined here, with 0 generic parameters
   --> /.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/block-padding-0.4.1/src/lib.rs:16:11
    |
 16 | pub trait Padding {
    |           ^^^^^^^

   Compiling hmac v0.13.0-rc.2
Some errors have detailed explanations: E0107, E0432.
For more information about an error, try `rustc --explain E0107`.
error: could not compile `cipher` (lib) due to 13 previous errors
warning: build failed, waiting for other jobs to finish...
```

--- a/picky-krb/Cargo.toml
+++ b/picky-krb/Cargo.toml
@@ -20,16 +20,24 @@ serde = { version = "1", features = ["derive"] }
 byteorder = "1.5"
 thiserror = "1"
 
-pbkdf2 = { version = "0.13.0-rc.1", features = ["sha1"] }
-hmac = "0.13.0-rc.1"
-sha1 = "0.11.0-rc.2"
+pbkdf2 = { version = "=0.13.0-rc.1", features = ["sha1"] }
+hmac = "=0.13.0-rc.2"
+sha1 = "=0.11.0-rc.2"
 
-aes = "0.9.0-rc.1"
-des = "0.9.0-rc.1"
-cbc = "0.2.0-rc.1"
-inout = "0.2.0-rc.6"
+aes = "=0.9.0-rc.1"
+des = "=0.9.0-rc.1"
+cbc = "=0.2.0-rc.1"
+inout = "=0.2.0-rc.6"
 rand = "0.9"
 
 uuid = { version = "1.18", default-features = false, features = ["serde"] }
 oid = "0.2"
-crypto-bigint = { version = "0.7.0-rc.6", features = ["alloc"] }
+crypto-bigint = { version = "=0.7.0-rc.8", features = ["alloc"] }
+
+# Pin transitive dependencies versions.
+# TODO: Remove when stable versions will be released.
+block-padding = "=0.4.0-rc.4"
+cipher = "=0.5.0-rc.1"
+block-buffer = "=0.11.0-rc.5"
+crypto-common = "=0.2.0-rc.4"
+digest = "=0.11.0-rc.3"

--- a/picky/Cargo.toml
+++ b/picky/Cargo.toml
@@ -47,35 +47,64 @@ reqwest = { version = "0.12", default-features = false, features = ["blocking"],
 
 rand = "0.9"
 rand_core = "0.9"
-crypto-bigint = "0.7.0-rc.6"
+crypto-bigint = "=0.7.0-rc.8"
 
-ed25519-dalek = { version = "3.0.0-pre.1", features = ["hazmat", "rand_core"] }
-x25519-dalek = { version = "3.0.0-pre.1", features = ["static_secrets"] }
+ed25519-dalek = { version = "=3.0.0-pre.1", features = ["hazmat", "rand_core"] }
+x25519-dalek = { version = "=3.0.0-pre.1", features = ["static_secrets"] }
 
-p256 = { version = "0.14.0-pre.11", features = ["ecdh"] }
-p384 = { version = "0.14.0-pre.11", features = ["ecdh"] }
-p521 = { version = "0.14.0-pre.11", features = ["ecdh"] }
+p256 = { version = "=0.14.0-pre.11", features = ["ecdh"] }
+p384 = { version = "=0.14.0-pre.11", features = ["ecdh"] }
+p521 = { version = "=0.14.0-pre.11", features = ["ecdh"] }
 
-rsa = { version = "0.10.0-rc.9", features = ["std"] }
+rsa = { version = "=0.10.0-rc.9", features = ["std"] }
 
-digest = "0.11.0-rc.3"
-md5 = { package = "md-5", version = "0.11.0-rc.2", features = ["oid"] }
-sha1 = { version = "0.11.0-rc.2", features = ["oid"] }
-sha2 = { version = "0.11.0-rc.2", features = ["oid"] }
-sha3 = { version = "0.11.0-rc.3", features = ["oid"] }
+digest = "=0.11.0-rc.3"
+md5 = { package = "md-5", version = "=0.11.0-rc.2", features = ["oid"] }
+sha1 = { version = "=0.11.0-rc.2", features = ["oid"] }
+sha2 = { version = "=0.11.0-rc.2", features = ["oid"] }
+sha3 = { version = "=0.11.0-rc.3", features = ["oid"] }
 
-aes-gcm = { version = "0.11.0-rc.1", optional = true }
-aes = { version = "0.9.0-rc.1", optional = true }
-aes-kw = { version = "0.3.0-rc.1", optional = true }
-argon2 = { version = "0.6.0-rc.1", optional = true }
-ctr = { version = "0.10.0-rc.1", optional = true }
-cbc = { version = "0.2.0-rc.1", optional = true, features = ["alloc"] }
-bcrypt-pbkdf = { version = "0.11.0-rc.1", optional = true }
-des = { version = "0.9.0-rc.1", optional = true }
-rc2 = { version = "0.9.0-pre.0", optional = true }
-pbkdf2 = { version = "0.13.0-rc.1", optional = true }
-hmac = { version = "0.13.0-rc.1", optional = true }
-inout = "0.2.0-rc.6"
+aes-gcm = { version = "=0.11.0-rc.1", optional = true }
+aes = { version = "=0.9.0-rc.1", optional = true }
+aes-kw = { version = "=0.3.0-rc.1", optional = true }
+argon2 = { version = "=0.6.0-rc.1", optional = true }
+ctr = { version = "=0.10.0-rc.1", optional = true }
+cbc = { version = "=0.2.0-rc.1", optional = true, features = ["alloc"] }
+bcrypt-pbkdf = { version = "=0.11.0-rc.1", optional = true }
+des = { version = "=0.9.0-rc.1", optional = true }
+rc2 = { version = "=0.9.0-pre.0", optional = true }
+pbkdf2 = { version = "=0.13.0-rc.1", optional = true }
+hmac = { version = "=0.13.0-rc.2", optional = true }
+inout = "=0.2.0-rc.6"
+
+# Pin transitive dependencies versions.
+# TODO: Remove when stable versions will be released.
+aead = { version = "=0.6.0-rc.2", optional = true }
+blake2 = { version = "=0.11.0-rc.2", optional = true }
+block-padding = "=0.4.0-rc.4"
+block-buffer = "=0.11.0-rc.5"
+blowfish = { version = "=0.10.0-rc.1", optional = true }
+cipher = { version = "=0.5.0-rc.1", optional = true }
+crypto-common = "=0.2.0-rc.4"
+ghash = { version = "=0.6.0-rc.2", optional = true }
+polyval = { version = "=0.7.0-rc.2", optional = true }
+universal-hash = { version = "=0.6.0-rc.2", optional = true }
+password-hash = { version = "=0.6.0-rc.1", optional = true }
+ed25519 = "=3.0.0-rc.1"
+signature = "=3.0.0-rc.4"
+ecdsa = "=0.17.0-rc.7"
+der = "=0.8.0-rc.9"
+pem-rfc7468 = "=1.0.0-rc.3"
+elliptic-curve = "=0.14.0-rc.15"
+hkdf = "=0.13.0-rc.2"
+pkcs8 = "=0.11.0-rc.7"
+spki = "=0.8.0-rc.4"
+sec1 = "=0.8.0-rc.10"
+rfc6979 = "=0.5.0-rc.1"
+pkcs1 = "=0.8.0-rc.4"
+keccak = "=0.2.0-rc.0"
+primefield = "=0.14.0-pre.6"
+primeorder = "=0.14.0-pre.9"
 
 [dev-dependencies]
 pretty_assertions = "1.4"
@@ -92,9 +121,15 @@ default = ["x509", "jose", "http_signature", "http_trait_impl", "pkcs12"]
 
 # main features
 x509 = []
-jose = ["dep:serde_json", "dep:aes-gcm", "dep:cbc", "dep:aes-kw"]
+jose = ["dep:serde_json", "dep:aes-gcm", "dep:cbc", "dep:aes-kw",
+# Transitive dependencies
+    "dep:aead", "dep:aes", "dep:cipher", "dep:ctr", "dep:ghash", "dep:polyval", "dep:universal-hash"
+]
 http_signature = []
-pkcs12 = ["picky-asn1-x509/pkcs12", "dep:des", "dep:rc2", "dep:cbc", "dep:pbkdf2", "dep:hmac", "dep:aes", "x509"]
+pkcs12 = ["picky-asn1-x509/pkcs12", "dep:des", "dep:rc2", "dep:cbc", "dep:pbkdf2", "dep:hmac", "dep:aes", "x509",
+# Transitive dependencies
+    "dep:cipher"
+]
 
 # secondary features
 pkcs7 = ["x509", "picky-asn1-x509/pkcs7"]
@@ -102,11 +137,17 @@ http_timestamp = ["dep:reqwest"]
 ctl = ["picky-asn1-x509/ctl", "pkcs7", "chrono_conversion"]
 ctl_http_fetch = ["dep:reqwest", "dep:cab", "ctl"]
 wincert = ["x509", "dep:byteorder"]
-ssh = ["dep:byteorder", "dep:aes", "dep:ctr", "dep:cbc", "dep:bcrypt-pbkdf", "dep:lexical-sort"]
+ssh = ["dep:byteorder", "dep:aes", "dep:ctr", "dep:cbc", "dep:bcrypt-pbkdf", "dep:lexical-sort",
+# Transitive dependencies
+    "dep:cipher", "dep:blowfish", "dep:pbkdf2", "dep:hmac"
+]
 http_trait_impl = ["dep:http"]
 chrono_conversion = ["dep:chrono", "picky-asn1/chrono_conversion"]
 time_conversion = ["dep:time", "picky-asn1/time_conversion"]
-putty = ["dep:argon2", "dep:hmac", "ssh"]
+putty = ["dep:argon2", "dep:hmac", "ssh",
+# Transitive dependencies
+    "dep:blake2", "dep:password-hash"
+]
 
 [package.metadata.docs.rs]
 # Enable all features when building documentation for docs.rs

--- a/picky/src/signature.rs
+++ b/picky/src/signature.rs
@@ -3,7 +3,6 @@
 use crate::hash::HashAlgorithm;
 use crate::key::ec::{EcComponent, EcCurve, NamedEcCurve};
 use crate::key::{KeyError, PrivateKey, PublicKey};
-use ed25519_dalek::ed25519::SignatureEncoding;
 use rsa::signature::Signer;
 
 use picky_asn1_x509::{AlgorithmIdentifier, oids};

